pkg/controller/atlas/provider.go

```diff
@@ -111,6 +113,46 @@ func (p *ProductionProvider) Client(ctx context.Context, secretRef *client.Objec
 	return c, secretData[orgIDKey], err
 }
 
+func (p *ProductionProvider) SdkClient(ctx context.Context, secretRef *client.ObjectKey, log *zap.SugaredLogger) (*admin.APIClient, string, error) {
+	secretData, err := getSecrets(ctx, p.k8sClient, secretRef, &p.globalSecretRef)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// TODO review we need add a custom logger to http client
+	//httpClientWithCustomLogger := http.DefaultClient
+	//err = httputil.LoggingTransport(log)(http.DefaultClient)
+	//if err != nil {
+	//	return nil, "", err
+	//}
+
+	c, err := NewClient(p.domain, secretData[publicAPIKey], secretData[privateAPIKey])
+
+	return c, secretData[orgIDKey], err
+}
+
+func getSecrets(ctx context.Context, k8sClient client.Client, secretRef, fallbackRef *client.ObjectKey) (map[string]string, error) {
```

Comment:
why not return a explicit struct with the fields we use for the secret instead of a map?